lib/charms/grafana_k8s/v0/grafana_metadata.py

```diff
@@ -0,0 +1,250 @@
+"""grafana_metadata.
+
+This implements provider and requirer sides of the grafana-metadata interface, which is used to communicate information
+about an Grafana installation such as its url and UID.
```

Comment:
```suggestion
about a Grafana installation such as its internal and external urls and UID.
```

I'd like to see clearly stated somewhere that the semantics of the interface is: "who am I, and how do you talk to me"

Reply:
also, is this meant to be scalable to applications that have multiple queriable urls?
If one day we solve scaling up grafana, could there be urls for multiple units in there?

Reply:
If one day we solve scaling up Grafana, we would presumably switch to ingress-per-app right? At that point you would still want only one URL.

But that does bring up the question of what internal address to supply when we scale up grafana now. Do we need to supply the leaders address rather than the service address?

Reply:
@PietroPasotti do you mean that we should publish this in charm relation interfaces (and can add a link in the docstring), or something more detailed here?  I'm +1 for that

@dstathis I think we've talked about how we wouldn't scale up grafana (because we'd charm [insert app that I dont remember the name of] instead?)